elide-async/src/main/java/com/yahoo/elide/async/export/formatter/JSONExportFormatter.java

```diff
@@ -38,26 +49,49 @@ public String format(PersistentResource resource, Integer recordNumber) {
             str.append(COMMA);
         }
 
-        str.append(resourceToJSON(mapper, resource));
+        str.append(resourceToJSON(jsonMapper, resource));
         return str.toString();
     }
 
-    public static String resourceToJSON(ObjectMapper mapper, PersistentResource resource) {
+    public static String resourceToJSON(JsonApiMapper jsonMapper, PersistentResource resource) {
         if (resource == null || resource.getObject() == null) {
             return null;
         }
 
         StringBuilder str = new StringBuilder();
-
         try {
-            str.append(mapper.writeValueAsString(resource.getObject()));
+            JsonApiDocument jsonApiDocument = new JsonApiDocument();
+            //TODO Make this a document processor
+            Data<Resource> data = resource == null ? null
+                    : new Data<>(resource.toResource(getRelationships(resource), getAttributes(resource)));
```

Comment:
`resource.toResource()` relies on the provided implementation of the getAttributes method in PersistenResource. It pulls the list of the Attributes from the dictionary which means it will have all the columns and not only the ones requested by the user. It does have some logic based on sparseFields but sparseFields is not populated in case of Graphql requests.